drivers/media/platform/raspberrypi/rp1_cfe/cfe.c

```diff
@@ -982,6 +983,35 @@ static void cfe_buffer_queue(struct vb2_buffer *vb)
 	spin_unlock_irqrestore(&cfe->state_lock, flags);
 }
 
+static u64 sensor_link_frequency(struct cfe_device *cfe)
+{
+	struct v4l2_mbus_framefmt *source_fmt;
+	struct v4l2_subdev_state *state;
+	const struct cfe_fmt *fmt;
+	s64 link_freq;
+
+	state = v4l2_subdev_lock_and_get_active_state(&cfe->csi2.sd);
+	source_fmt = v4l2_subdev_get_pad_format(&cfe->csi2.sd, state,
+						cfe->fe_csi2_channel);
+	fmt = find_format_by_code(source_fmt->code);
+	link_freq = v4l2_get_link_freq(cfe->sensor->ctrl_handler,
+				       fmt->depth,
+				       cfe->csi2.active_data_lanes * 2);
+	if (link_freq < 0) {
```

Comment:
Do we want any bounds checks here? ISTR the legal range is 80Mbps to 1.5Gbps. But perhaps that's too many magic numbers.

Reply:
There are further bounds checking when we go to program the DPHY frequency from ``csi2_open_rx()``.